store/views.py

```diff
@@ -77,6 +105,60 @@ def loanBookView(request):
 @csrf_exempt
 @login_required
 def returnBookView(request):
-    pass
+    response_data = {
+        'message': None,
+    }
+   
+    book_id = request.POST.get('bid')
+
+    try:
+        book_copy = BookCopy.objects.get(id=book_id)
+        response_data['message'] = 'success'
+        book_copy.borrower = None
+        book_copy.borrow_date = None
+        book_copy.status = True
+        book_copy.save()
+    except:
+        response_data['message'] = 'failure'
+        
+
+    return JsonResponse(response_data)
+
+
+@csrf_exempt
+@login_required
+def rateBookView(request):
+    response_data = {
+        'message': None,
+    }
+
+    if request.POST:
+        book_id = request.POST.get('bid')
+        book_rate = request.POST.get('brate')
+
+        books = Book.objects.get(id=book_id)
+        rating1 = BookRating.objects.filter(book=books, user=request.user)
+
+        if len(rating1)==0:
+            new_rating = BookRating(book=books, user=request.user, rating=book_rate)
+            new_rating.save()
+        else:
+            for i in rating1:
+                i.rating = book_rate
+                i.save()
+
+        Calc_Rate=BookRating.objects.filter(book=books)
+        book_rate = 0
+        for r in Calc_Rate:
+            book_rate +=  r.rating
+        book_rate /= len(Calc_Rate)
+        book_rate = round(book_rate,2)
```

Comment:
A very clean and more Django-ish way for this could be, using aggregate functions in Django ORM.
```python
book_rate = BookRating.objects.filter(book__pk=book_id).aggregate(Avg('rating'))['rating__avg']
```